sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -145,10 +146,142 @@ class ChatCompletionsClient(ChatCompletionsClientGenerated):
     :paramtype api_version: str
     """
 
-    def __init__(self, endpoint: str, credential: Union[AzureKeyCredential, "TokenCredential"], **kwargs: Any) -> None:
-        self._model_info: Optional[_models.ModelInfo] = None
+    def __init__(
+        self, endpoint: str, credential: Union[AzureKeyCredential, "TokenCredential"], **kwargs: Any
+    ) -> None:
+        self._endpoint = endpoint
+        self._credential = credential
+        self._init_kwargs = kwargs
+        self._model_info: Optional[_models.ModelInfo] = kwargs.get("model_info")
+        self._model_extras: Optional[Dict[str, Any]] = kwargs.get("model_extras")
+        self._frequency_penalty: Optional[float] = kwargs.get("frequency_penalty")
+        self._presence_penalty: Optional[float] = kwargs.get("presence_penalty")
+        self._temperature: Optional[float] = kwargs.get("temperature")
+        self._top_p: Optional[float] = kwargs.get("top_p")
+        self._max_tokens: Optional[int] = kwargs.get("max_tokens")
+        self._response_format: Optional[
+            Union[str, _models.ChatCompletionsResponseFormat]
+        ] = kwargs.get("response_format")
+        self._stop: Optional[List[str]] = kwargs.get("stop")
+        self._tools: Optional[List[_models.ChatCompletionsToolDefinition]] = kwargs.get("tools")
+        self._tool_choice: Optional[
+            Union[str, _models.ChatCompletionsToolSelectionPreset, _models.ChatCompletionsNamedToolSelection]
+        ] = kwargs.get("tool_choice")
+        self._seed: Optional[int] = kwargs.get("seed")
+        self._model: Optional[str] = kwargs.get("model")
```

Comment:
I think we need to re-order this to first kwargs.**pop**("option", None) all the options and then set `self._init_kwargs = kwargs`. Otherwise we could get duplicates in the call to with_defaults when we pass `**self._init_kwargs`. Consider this scenario:

```python
client = ChatCompletionsClient(
    endpoint="endpoint",
    credential=AzureKeyCredential("key"),
).with_defaults(
    temperature=0.1
)

new_client = client.with_defaults(
    temperature=0.1
)
```

> TypeError: azure.ai.inference._patch.ChatCompletionsClient() got multiple values for keyword argument 'model_info'

Reply:
Good call @kristapratico, thank you for the thorough code review! my existing tests never call `.with_defaults()` twice, so I never hit this. Fixing the code now and adding more tests.

Reply:
Fixed. Thanks!